int8 type
int8 in_mode
---
int8 out_mode1
int8 out_mode2
int8 out_mode3
